packages/blade/src/components/Toast/__tests__/Toast.test.stories.tsx

```diff
@@ -0,0 +1,237 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable import/no-extraneous-dependencies */
+import type { StoryFn } from '@storybook/react';
+import { within, userEvent } from '@storybook/testing-library';
+import { expect, jest } from '@storybook/jest';
+import React from 'react';
+import type { ToastProps } from '../types';
+import { useToast } from '../useToast';
+import { ToastContainer } from '../ToastContainer';
+import type { Toast } from '../Toast';
+import { Button } from '~components/Button';
+import { Box } from '~components/Box';
+
+const onDismissButtonClick = jest.fn();
+const ToastExample = (props: ToastProps): React.ReactElement => {
+  const toast = useToast();
+
+  React.useEffect(() => {
+    toast.dismiss();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <Box>
+      <Button
+        onClick={() => {
+          toast.show({ ...props });
+        }}
+      >
+        Show Toast
+      </Button>
+      <ToastContainer />
+    </Box>
+  );
+};
+
+const sleep = (ms: number): Promise<void> => new Promise((resolve) => setTimeout(resolve, ms));
+
+export const TestToastShow: StoryFn<typeof Toast> = (): React.ReactElement => {
+  return <ToastExample content="Payment successful" />;
+};
+
+TestToastShow.play = async () => {
+  const { getByRole, queryByText } = within(document.body);
+  await sleep(1000);
+
+  const toastContent = 'Payment successful';
+  await expect(queryByText(toastContent)).not.toBeInTheDocument();
+  const button = getByRole('button', { name: 'Show Toast' });
+  await userEvent.click(button);
+  await sleep(400);
```

Comment:
`await waitFor(() => )` instead of sleep

Reply:
Tried it before, doesn't work. We need to discreetly wait for the animations to finish before asserting.

Reply:
The exit would depend on absolute duration no? but I am guessing the toast opens immediately on button click. So can we not use waitFor on those parts? and keep sleep where we want to actually wait for some duration for toast to disappear etc?

Reply:
> So can we not use waitFor on those parts

Maybe somethings wrong with storybook testing lib, it doesn't work with waitFor. You need to wait for a discreet amount of time.